src/Illuminate/Database/Query/Builder.php

```diff
@@ -2183,6 +2183,42 @@ public function dynamicWhere($method, $parameters)
         return $this;
     }
 
+    /**
+     * Handles dynamic "where(Any|All|None)*" clauses to the query.
+     *
+     * @param  array  $matches
+     * @param  array  $parameters
+     * @return $this
+     *
+     * @throws \BadMethodCallException
+     */
+    public function dynamicWhereColumns($matches, $parameters)
+    {
+        $whereColumnsClause = $matches[3];
+        $boolean = (strtolower($matches[1]) === 'or') ? 'or' : 'and';
```

Comment:
When called from `__call` the matches has `/i` modifier so using `'or'` may exclude some matches. Also with the following `None` and `All` checks.

Reply:
Thanks for the review. I updated the code based on you input, added some inline comments to make the code clearer, and as well added a test to verify that it is working with lowercase method name